ci/docker/ubuntu-22.04-cpp.dockerfile

```diff
@@ -197,3 +196,7 @@ ENV absl_SOURCE=BUNDLED \
     PATH=/usr/lib/ccache/:$PATH \
     PYTHON=python3 \
     xsimd_SOURCE=BUNDLED
+
+# GH-36013 disabling opentelemetry here because we can't
+# get the patched version from conda
+#    ARROW_WITH_OPENTELEMETRY=ON \
```

Comment:
We can use comment in `ENV`:

```diff
diff --git a/ci/docker/ubuntu-22.04-cpp.dockerfile b/ci/docker/ubuntu-22.04-cpp.dockerfile
index e6fd44ff2..6dc5ccbcb 100644
--- a/ci/docker/ubuntu-22.04-cpp.dockerfile
+++ b/ci/docker/ubuntu-22.04-cpp.dockerfile
@@ -184,7 +184,9 @@ ENV absl_SOURCE=BUNDLED \
     ARROW_WITH_BROTLI=ON \
     ARROW_WITH_BZ2=ON \
     ARROW_WITH_LZ4=ON \
-    ARROW_WITH_OPENTELEMETRY=ON \
+    # GH-36013 disabling opentelemetry here because we can't
+    # get the patched version from conda
+    # ARROW_WITH_OPENTELEMETRY=ON \
     ARROW_WITH_SNAPPY=ON \
     ARROW_WITH_ZLIB=ON \
     ARROW_WITH_ZSTD=ON \
```

Reply:
Ah, thanks